xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -1066,6 +1066,12 @@ int CGUIDialogVideoInfo::ManageVideoItem(const CFileItemPtr &item)
         result = RemoveItemsFromTag(item);
         break;
 
+      case CONTEXT_BUTTON_MARK_WATCHED:
+      case CONTEXT_BUTTON_MARK_UNWATCHED:
+        database.MarkAsWatched(*item, (button == (CONTEXT_BUTTON)CONTEXT_BUTTON_MARK_WATCHED));
```

Comment:
Btw, having two places for one button is still weird and should be avoided ;) But that's another discussion.